apiserver/common/crossmodelcommon/crossmodel.go

```diff
@@ -82,6 +108,17 @@ func (api *BaseAPI) ApplicationOffersFromModel(
 
 	var results []params.ApplicationOfferDetails
 	for _, offer := range offers {
+		// If the user is not a model admin, they need at least read
+		// access on an offer to see it.
+		if !isAdmin {
+			permErr := api.CheckPermission(names.NewApplicationOfferTag(offer.OfferName), permission.ReadAccess)
+			if permErr == common.ErrPerm {
+				continue
+			}
+			if err != nil {
```

Comment:
This should be `if permErr != nil`, shouldn't it? (And `errors.Trace(permErr)` below.)